pkgs/development/compilers/swift/default.nix

```diff
@@ -120,6 +120,9 @@ let
   builder = ''
     # gcc-6.4.0/include/c++/6.4.0/cstdlib:75:15: fatal error: 'stdlib.h' file not found
     NIX_CFLAGS_COMPILE="$( echo ${clang.default_cxx_stdlib_compile} ) $NIX_CFLAGS_COMPILE"
```

Comment:
Whhhyyyyyyy

1) Do we really not build with clang? :cry:
2) WTF putting CXXFLAGS in CFLAGS, yikes
3) WTF the mix of headers, compilers, runtime libraries-- do we even know what combination is used? Is it consistent throughout the various builds?!

..... And this isn't even something you're adding, so don't worry about it.  But... :(. D'oh.

Reply:
Yes, this is pretty gnarly. The way I see it, we do indeed use clang, but clang has gcc as a dependency and uses some of its compiler-shipping headers and startup files. The need for these extra flags however is a bit weird for me as well. I tried twice (but just briefly) to get rid of it, because I get bad vibes from it too, but so far failed to diagnose the actual problem. Nix’s clang should be able to find its stuff without help, but somehow it does not.